File: src/main/resources/static/js/work.js
class Work{

    url
    title
    authorData=[]
    node
    banner
    icon
    location ={}
    tags = []
    language = []
    beginLinks = []
    instructions
    year
    techDetails
    requirements = []
    artist = {}
    editorialStatement
    sourceFiles
    video
    externalUrl
    copyright
    previousPublication = []
    references = []

    constructor(fields,tags) {
        this.tags=tags;
        this.populateFromFields(fields);
    }

    populateFromFields(fields){
        let author={};
        for(let i=0;i<$(fields).length;i++){
            let field=$(fields)[i];
            let id=$(field).attr('id');
            let value=$(field).val();

            if(id==='authorFirst'){
                author.first=value;
            }else if(id==='authorLast'){
                author.last=value;
                this.authorData.push(author);
                author={};
            }
            else if(id==='requirements'){
                this.requirements=value.split(',');
            }
            else if(id==='previousPublication'){
                this.previousPublication=value.split(',');
            }
            else if(id==='references'){
                this.references=value.split(',');
            }
            else if(id==='beginLinks'){
                let beginLink={}
                beginLink.displayText='Begin';
                beginLink.link=value;
                this.beginLinks.push(beginLink);
            }else if(id==='city'){
                this.location.city=value;
            }else if(id==='country'){
                this.location.country=value;
            }else if(id==='language'){
                this.language=value.split(',');
            }
            else{
                this[id]=value;
            }
        }
    }
}

File: src/main/java/pt/ufp/lpi/backofficerui/models/Work.java
package pt.ufp.lpi.backofficerui.models;

import com.fasterxml.jackson.annotation.JsonIgnore;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.ArrayList;
import java.util.Comparator;
import java.util.List;

@Data
class Autor {
    private String first;
    private String last;
}
@Data
class Location{
    private String city;
    private String country;
}
@Data
class Link{
    private String displayText;
    private String link;
}
@Data
class Artist{
    private String statement;
    private String bio;
}

@Data
@NoArgsConstructor
@AllArgsConstructor
public class Work {
    private String url;
    private String title;
    private String authorDisplayName;
    private List<Autor> authorData=new ArrayList<>();
    private String node;
    private String banner;
    private String icon;
    private Location location;
    private final List<String> tags=new ArrayList<>();
    private final List<String> language=new ArrayList<>();
    private final List<Link> beginLinks=new ArrayList<>();
    private String instructions;
    private String year;
    private String techDetails;
    private final List<String> requirements=new ArrayList<>();
    private Artist artist;
    private String editorialStatement;
    private String sourceFiles;
    private String video;
    private String externalUrl;
    private String copyright;
    private final List<String> previousPublication=new ArrayList<>();
    private final List<String> references=new ArrayList<>();
}
